src/typescript/azure-openapi-validator/rules/PageableOperation.ts

```diff
@@ -0,0 +1,39 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { MergeStates, OpenApiTypes, rules } from '../rule';
+export const PageableOperation: string = "PageableOperation";
+
+const jp = require('jsonpath');
+
+rules.push({
+  id: "R2029",
+  name: PageableOperation,
+  severity: "warning",
+  category: "SDKViolation",
+  mergeState: MergeStates.composed,
+  openapiType: OpenApiTypes.arm | OpenApiTypes.dataplane,
+  appliesTo_JsonQuery: "$.paths[?(@.get)]",
+  run: function* (doc, node, path) {
+    const operations = Object.keys(node);
+    const getKey = operations.find(key => {
+      return key.toLowerCase() === 'get';
+    });
+
+    const schemaRef = node[getKey]['responses']['200']['schema']['$ref'];
+    const schemaPath: string[] = (<string>schemaRef).split('/');
+    const schemaProperties = doc.definitions[schemaPath[2]].properties;
```

Comment:
What will happen if the model is defined inline?

Reply:
I think we can run with this assumption since there are linter rules in place to flag inline definitions
On the other hand though, moving this to a utility method could be helpful